_logistic_regression.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import annotations
+
+from typing import Callable, Tuple
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.linear_model import LogisticRegression as mvLogisticRegression
+from sklearn.utils.validation import check_is_fitted
+
+from ..._utils import _classifier_get_classes
+from ...representation import FDataGrid
+from ...representation._typing import NDArrayAny, NDArrayInt
+
+
+class LogisticRegression(
+    BaseEstimator,  # type: ignore
+    ClassifierMixin,  # type: ignore
+):
+    r"""Logistic Regression classifier for functional data.
+
+    This class implements the sequential “greedy” algorithm
+    for functional logistic regression proposed in
+    :footcite:ts:`bueno+larraz_2021_functional`.
+
+    .. warning::
+        For now, only binary classification for functional
+        data with one dimensional domains is supported.
+
+    Args:
+        p: number of points (and coefficients) to be selected by
+        the algorithm.
+
+    Attributes:
+        classes\_: A list containing the name of the classes
+        points\_: A list containing the selected points.
+        coef\_: A list containing the coefficient for each selected point.
+        intercept\_: Independent term.
+
+    Examples:
+        >>> from numpy import array
+        >>> from skfda.datasets import make_gaussian_process
+        >>> from skfda.ml.classification import LogisticRegression
+        >>> fd1 = make_gaussian_process(
+        ...         n_samples=50,
+        ...         n_features=100,
+        ...         noise=0.5,
+        ...         random_state=0,
+        ... )
+        >>> fd2 = make_gaussian_process(
+        ...         n_samples=50,
+        ...         n_features = 100,
+        ...         mean = np.array([1]*100),
+        ...         noise = 0.5,
+        ...         random_state=0
+        ... )
+        >>> fd = fd1.concatenate(fd2)
+        >>> y = 50*[0] + 50*[1]
+        >>> lr = LogisticRegression(p=2)
+        >>> _ = lr.fit(fd[::2], y[::2])
+        >>> lr.coef_.round(2)
+        array([[ 2.41,  1.68]])
+        >>> lr.points_.round(2)
+        array([ 0.11,  0.  ])
+        >>> lr.score(fd[1::2],y[1::2])
+        0.92
+
+        References:
+            .. footbibliography::
+
+    """
+
+    def __init__(
+        self,
+        p: int = 5,
+    ) -> None:
+
+        self.p = p
+
+    def fit(  # noqa: D102
+        self,
+        X: FDataGrid,
+        y: NDArrayAny,
+    ) -> LogisticRegression:
+
+        X, classes, y_ind = self._argcheck_X_y(X, y)
+
+        self.classes_ = classes
+
+        n_samples = len(y)
+        n_features = len(X.grid_points[0])
+
+        selected_indexes = np.zeros(self.p, dtype=np.intc)
+
+        mvlr = mvLogisticRegression()  # multivariate logistic regression
+
+        x_mv = np.zeros((n_samples, self.p))
+        LL = np.zeros(n_features)
+        for q in range(self.p):
+            for t in range(n_features):
```

Comment:
This is very slow, right? Couldn't `LL` be considered a continuous function of `t`, and then use optimization techniques for functions, so that we don't need to evaluate it for every `t`? Is that considered anywhere in the literature?

Reply:
I thought about using optimization techniques for `LL` as well, but I discarded that option because in this algorithm you have to fit the parameters for each `t`.

Reply:
I am not sure why it should be a problem. My proposal is to consider that the thing computed in the inner loop is a continuous function of `t` (please, check that it is indeed continuous). If that is the case, computing it for values of `t` that are nearby is wasteful, as the value of `LL` will not change much.

Thus, we could use any of the methods in `https://docs.scipy.org/doc/scipy/reference/optimize.html#global-optimization` to optimize that function. In addition, this extends the applicability of this method to basis representation, and multiple domain dimensions as it only requires function evaluation.

I think we should discuss this in the next meeting.

Reply:
Okay, I think I understand now. 

> I think we should discuss this in the next meeting.
Perfect.

Reply:
Okay, I think I understand now.

> I think we should discuss this in the next meeting.

Perfect.